lib/tracing.py

```diff
@@ -0,0 +1,168 @@
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+   Make an as_span decorator to help us log traces with a jaeger trace service
+"""
+
+from functools import wraps
+import os
+import sys
+import logging
+from typing import Dict, Any, Callable, TypeVar, Optional, List
+
+# opentelemetry makes this obnoxious warning about nanosecond accuracy in python 3.6
+# squelch it before we import the modules
+class nanosecond_warning_filter(logging.Filter):
+    def filter(self, lr: logging.LogRecord) -> bool:
+        if str(lr).find("millisecond precision") > 0:
+            return False
+        return True
+
+
+logging.getLogger("opentelemetry.util._time").addFilter(nanosecond_warning_filter())
+
+try:
+    # from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter # type: ignore
+    # from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter # type: ignore
+    # pylint: disable-next=wrong-import-position
+    from opentelemetry.exporter.jaeger.thrift import JaegerExporter  # type: ignore
+    from opentelemetry.sdk.resources import Resource  # type: ignore
+    from opentelemetry.sdk.trace import TracerProvider  # type: ignore
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor  # type: ignore
+    from opentelemetry import trace  # type: ignore
+    from opentelemetry.context import Context  # type: ignore
+    from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator  # type: ignore
+
+    resource = Resource(attributes={"service.name": "fife"})
+
+    trace.set_tracer_provider(TracerProvider(resource=resource))
+
+    # otlp_exporter = OTLPSpanExporter(
+    #    endpoint="https://landscape.fnal.gov/jaeger-collector/api/traces"
+    # )
+    # span_processor = BatchSpanProcessor(otlp_exporter)
+
+    jaeger_exporter = JaegerExporter(
+        collector_endpoint=os.environ["OTEL_EXPORTER_JAEGER_ENDPOINT"]
+    )
+    span_processor = BatchSpanProcessor(jaeger_exporter)
+
+    trace.get_tracer_provider().add_span_processor(span_processor)
+
+    tracer = trace.get_tracer("jobsub_lite")
+
+    def get_current_span():  # type: ignore
+        return trace.get_current_span()
+
+    def get_propagator_carrier() -> Dict[str, str]:
+        carrier: Dict[str, str] = {}
+        TraceContextTextMapPropagator().inject(carrier)
+        return carrier
+
+except:
+    # if we can't import the stuff, here's a little mock so we don't crash
+    print("Note: tracing not available here.")
```

Comment:
Could you please add to this message that we'll still continue the operation without tracing?  The first time I saw this, I Ctrl-C'd because I thought it wouldn't work.

Reply:
I love the fact that it still DOES work due to your mock.  That's a great touch.